src/os/i18n.zig

```diff
@@ -108,12 +108,15 @@ pub fn canonicalizeLocale(
     // Buffer must be 16 or at least as long as the locale and null term
     if (buf.len < @max(16, locale.len + 1)) return error.NoSpaceLeft;
 
+    // Fix zh locales for macOS
+    const updated = fixZhLocale(locale);
```

Comment:
Is there even a point of canonicalizing via libintl for "fixed" zh locales? If not then we can possibly have something like this: (the return type would need to be changed to `?[:0]const u8`)
```suggestion
    if (fixZhLocale(locale)) |fixed| {
        return fixed;
    } else {
        // normal logic
    }
```